src/AutoRest.CSharp/Common/Output/Models/RestClientBuilder.cs

```diff
@@ -64,8 +67,17 @@ private string GetRequestParameterName (RequestParameter requestParameter)
         public RestClientMethod BuildMethod(Operation operation, HttpRequest httpRequest, IEnumerable<RequestParameter> requestParameters, DataPlaneResponseHeaderGroupType? responseHeaderModel, string accessibility)
         {
             Dictionary<RequestParameter, ConstructedParameter> allParameters = new();
-
-            List<RequestParameter> parameters = operation.Parameters.Concat(requestParameters).ToList();
+            //TODO: remove this part of code which makes subscriptionId and managementGroupId as the first parameter when parameters are sorted by their order in path.
+            var isAncestorResourceTypeTenant = false;
+            try
+            {
+                isAncestorResourceTypeTenant = _operationGroup.IsAncestorResourceTypeTenant(_context);
+            }
+            catch (ArgumentException)
+            {
+                // do nothing for exception in dataplane
+            }
+            List<RequestParameter> parameters = (isAncestorResourceTypeTenant ? operation.Parameters.OrderBy(p => !p.Language.Default.Name.Equals("subscriptionId", StringComparison.InvariantCultureIgnoreCase) && !p.Language.Default.Name.Equals("managementGroupId", StringComparison.InvariantCultureIgnoreCase)).ToList() : operation.Parameters).Concat(requestParameters).ToList();
```

Comment:
The workaround can be removed after #1423 is merged.

Reply:
This pr should have covered it https://github.com/Azure/autorest.csharp/pull/1431 instead of 1423

Reply:
Code removed.